### PR TITLE
Add Spotless with ktlint to the project

### PR DESCRIPTION
**Pull request by yasith**
Run using 
./gradlew spotlessCheck --init-script gradle/init.gradle.kts --stacktrace

Apply automatic changes using
./gradlew spotlessApply --init-script gradle/init.gradle.kts --stacktrace

### CODE REVIEW

**Comment by yasith** [comment]
Current dependencies on/for this PR:
* main
  * **PR #52** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/52" target="_blank">Graphite</a>  👈
    * **PR #51** <a href="https://app.graphite.dev/github/pr/google/jetpack-camera-app/51" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/google/jetpack-camera-app/52?utm_source=stack-comment).

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[*.{kt,kts}]
+ktlint_standard_function-naming = disabled
+#ktlint_function_naming_ignore_when_annotated_with=Composable
```
Comment (temcguir) [inline .editorconfig:3]: Is this line needed? It doesn't seem to have caught any compose functions even with it commented out.

File: gradle/init.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+rootProject {
+    subprojects {
+        apply<com.diffplug.gradle.spotless.SpotlessPlugin>()
+        extensions.configure<com.diffplug.gradle.spotless.SpotlessExtension> {
+            kotlin {
+                target("**/*.kt")
+                targetExclude("**/build/**/*.kt")
+                ktlint(ktlintVersion)
+                    .userData(mapOf("android_studio" to "true"))
```
Comment (temcguir) [inline gradle/init.gradle.kts:39]: The [spotless documentation](https://github.com/diffplug/spotless/tree/main/plugin-gradle#ktlint) says to use `mapOf("android" to "true")`.

I can see in the ktlint documentation that `--android` was deprecated, but according to [this issue](https://github.com/pinterest/ktlint/issues/1982) it looks like we should be using `mapOf("code-style", "android_studio")` or maybe just adding `ktlint_code_style=android_studio` in `.editorconfig`.

**Review by temcguir — APPROVED** [review]
LGTM. Just a few questions.

File: .editorconfig (hunk 1 above)
Comment (yasith) [reply .editorconfig:3]: Ah sorry, I was experimenting with the snapshot version of ktlint. I opened a bug to switch to this after ktlint 1.0.1 is released.

I'll delete this line for now.

File: gradle/init.gradle.kts (hunk 2 above)
Comment (yasith) [reply gradle/init.gradle.kts:39]: That's a good spot, I'll move it to `.editorconfig` I think that's cleaner.

**Review by davidjiagoogle — APPROVED** [review]

**Comment by yasith** [comment]
### Merge activity

* **Oct 11, 4:26 PM**: @yasith started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/google/jetpack-camera-app/52).

**Pull request merged**